### PR TITLE
feat: Use native Promise, but allow overriding it

### LINKED ISSUE #560: Use native promises instead of bluebird
[Redis 4.0 has stopped using bluebird natively](https://github.com/luin/ioredis/blob/master/Changelog.md#400-2018-08-14). As a result, this code will work with ioredis-mock but fail with the real ioredis:

```JavaScript
return redisClient.hset(USERS_HASH_NAME, user.id, JSON.stringify(user))
  .thenReturn(user);
```

That was a confusing situation for me, as you can imagine!

ioredis 4 does offer support for bluebird promises, with a little setup:

```JavaScript
const Promise = require('bluebird');
const Redis = require('ioredis');

Redis.Promise = Promise;
```

### PR DESCRIPTION
**Pull request by ErinCall**
Fixes #560.

I've cribbed from [the corresponding changes in the ioredis library](https://github.com/luin/ioredis/commit/da60b8b); as a result it has the same API. So that's nice!

I _think_ it's still compatible with pre-Promise versions of node, as long as `Redis.Promise` gets set before trying to perform any operations. I don't have an older version on hand for testing, though.

### CODE REVIEW

**Comment by ErinCall** [comment]
> `error Your lockfile needs to be updated, but yarn was run with \`--frozen-lockfile\`.`

I'm not familiar with yarn; how can I resolve that error?

File: README.md [hunk 1]
```diff
@@ -40,6 +40,17 @@ var redis = new Redis({
 // Basically use it just like ioredis
 ```
 
+### Promises
+
+By default, ioredis-mock uses the native Promise library. If you need (or prefer) [bluebird](http://bluebirdjs.com/) promises, set `Redis.Promise`:
```
Comment (stipsan) [inline README.md]: Thanks for taking the time to update the readme 😄

**Comment by ErinCall** [comment]
Oh, I meant to ask: should there be changes to the version or the `ioredis` peer dependency? It seems like there probably should be, since it’s a breaking change and it’s meant to match ioredis 4 specifically.

**Comment by stipsan** [comment]
ah, good point! Yeah lets make it a major breaking change and update the peerDependency 😄

**Comment by ErinCall** [comment]
Hm, I guess I don't know how to make those changes. The peerDependency in package.json is simple enough to update, but after updating it and running `yarn`, `yarn info` still shows `'2.x || 3.x || 4.x'`. 

I don't see where to set the version at all. It looks like it might be based on the git tags, in which case I think that has to be done on your end?

**Comment by stipsan** [comment]
Yep the version is handled by [semantic-release](https://github.com/semantic-release/semantic-release) after merging the PR 😄

**Pull request merged**